os/net/lwip/src/include/lwip/lwipopts.h

```diff
@@ -652,15 +652,27 @@
 #define LWIP_RANDOMIZE_INITIAL_LOCAL_PORTS      CONFIG_NET_RANDOMIZE_INITIAL_LOCAL_PORTS
 #endif
 
-#ifdef CONFIG_NFILE_DESCRIPTORS
-#define LWIP_SOCKET_OFFSET CONFIG_NFILE_DESCRIPTORS
+#if !defined(CONFIG_NSOCKET_DESCRIPTORS) || (CONFIG_NSOCKET_DESCRIPTORS == 0)
+#error "CONFIG_NSOCKET_DESCRIPTORS is undefined or 0"
 #endif
 
-#ifdef CONFIG_NSOCKET_DESCRIPTORS
-#define MEMP_NUM_NETCONN CONFIG_NSOCKET_DESCRIPTORS
-#define MEMP_NUM_RAW_PCB CONFIG_NSOCKET_DESCRIPTORS
+#ifndef CONFIG_NFILE_DESCRIPTORS
+#define CONFIG_NFILE_DESCRIPTORS 0
 #endif
 
+#ifndef CONFIG_NUDS_DESCRIPTORS
+#define CONFIG_NUDS_DESCRIPTORS 0
+#endif
+
+#define LWIP_SOCKET_OFFSET (CONFIG_NFILE_DESCRIPTORS + CONFIG_NUDS_DESCRIPTORS)
+
+#define CONFIG_NLWIP_DESCRIPTORS (CONFIG_NSOCKET_DESCRIPTORS - CONFIG_NUDS_DESCRIPTORS)
+#if (CONFIG_NLWIP_DESCRIPTORS <= 0)
```

Comment:
it would be better to write the guide that nsocket_descriptors is consist of nlwip_descriptors and nuds_descriptors. by the way, CONFIG_NLWIP_DESCRIPTORS is not generated by Kconfig then I think you can remove CONFIG_ prefix

Reply:
I agree with your idea.
I think `CONFIG_NUDS_DESCRIPTOR` and `CONFIG_NLWIP_DESCRIPTOR` should be generated by Kconfig, and `CONFIG_NSOCKET_DESCRIPTOR` should be the sum of them.
But it is impossible that generating a new config by adding two configs.
How can I configure `CONFIG_NSOCKET_DESCRIPTOR`? Do you have any idea?